test/take.cc

```diff
@@ -0,0 +1,117 @@
+#include "stout/take.h"
+
+#include <vector>
+
+#include "gtest/gtest.h"
+#include "stout/collect.h"
+#include "stout/filter.h"
+#include "stout/iterate.h"
+#include "stout/terminal.h"
+
+using stout::eventuals::Collect;
+using stout::eventuals::Filter;
+using stout::eventuals::Iterate;
+using stout::eventuals::TakeFirstN;
+using stout::eventuals::TakeLastN;
+using stout::eventuals::TakeRange;
+
+TEST(Take, IterateTakeLastCollect) {
+  std::vector<int> v = {5, 12, 17, 3};
+
+  auto s = [&]() {
+    return Iterate(v)
+        | TakeLastN(2)
+        | Collect<std::vector<int>>();
+  };
+
+  auto res = *s();
+
+  EXPECT_EQ(17, res.at(0));
+  EXPECT_EQ(3, res.at(1));
+  EXPECT_EQ(2, res.size());
```

Comment:
Can you investigate how hard it would be to use [`googlemock`](https://github.com/google/googletest/tree/master/googlemock) in our tests? If that works, you can write this whole thing as:
```suggestion
  EXPECT_THAT(res, ElementsAre(17, 3));
```
... And it'll give you very nice errors if the test fails.

See [this StackOverflow answer](https://stackoverflow.com/questions/1460703/comparison-of-arrays-in-google-test) for some more examples of what you can do.